test/pylib/report_plugin.py

```diff
@@ -24,7 +24,8 @@ def pytest_configure(self, config):
     def pytest_runtest_makereport(self):
         outcome = yield
         report = outcome.get_result()
-        report.nodeid = f"{report.nodeid}.{self.mode}.{self.run_id}"
+        if self.mode is not None or self.run_id is not None:
+            report.nodeid = f"{report.nodeid}.{self.mode}.{self.run_id}"
```

Comment:
It's better to check each parameter separately, to avoid the case with None in the name when one of the parameters will not be passed.

Reply:
I started this with "and" instead of "or", but then I thought, maybe some crazy use case will want to set one but not the other, and if someone does this, you just get the word "NONE" instead of one of the components. I actually tested that.

In any case, nobody will do this thing... test.py will set both, and pytest or test/*/run will not set either.

Reply:
True, but you can check it one by one and add if it's set.

Reply:
And what - have test_name.1 or test_name.mode alone? Who wants that? I assumed that you wanted this test_name.mode.number format and didn't want one of them to be missing, so it's preferable to write NONE instead of nothing. But I don't know, since nobody will ever set just one of them :-) Your code sets both, and direct run of pytest sets neither.

Reply:
Could sidestep the question by folding the two parameters into a --label option or something.